stdlib/socket/src/main/ballerina/socket/listener_endpoint.bal

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents service endpoint where socket server service registered and start.
+#
+public type Listener object {
+
+    *AbstractListener;
+
+    public function __init(ListenerConfig config) {
```

Comment:
Change signature to return error? instead ?

Reply:
Listener object initialization will start the server socket. So if an error occurred then service deployment should be halted. So throwing panic should be the right approach IMO.